src/datasources/db/v2/database-migrator.service.ts

```diff
@@ -0,0 +1,157 @@
+
+    const connection =
+      await this.databaseService.initializeDatabaseConnection();
+    await this.createLockTableIfNotExists(connection);
+
+    let numberOfIterations = 0;
+    const numberOfRetries = await this.configService.getOrThrow(
+      'db.migrator.numberOfRetries',
+    );
+    while (numberOfRetries >= numberOfIterations) {
+      ++numberOfIterations;
+      if (!(await this.lockExists(connection))) {
+        await this.insertLock(connection);
+        await this.runMigration(connection);
+        await this.truncateLocks(connection);
+
+        this.loggingService.info('Migrations: Finished.');
+        break;
+      } else {
+        if (numberOfIterations === numberOfRetries) {
+          throw new Error(
+            'Migrations: Migrations are still running in another instance!',
+          );
+        }
+        this.loggingService.info('Migrations: Running in another instance...');
+        const retryAfter = this.configService.getOrThrow<number>(
+          'db.migrator.retryAfter',
+        );
+
+        await new Promise((resolve) => setTimeout(resolve, retryAfter));
+      }
+    }
+  }
+
+  /**
+   * Creates the locks table in the database if it does not already exist.
+   *
+   * @param {DataSource} connection The database connection to create the locks table on.
+   *
+   * @returns {Promise<void>} A promise that resolves when the locks table has been created or confirmed to exist.
+   */
+  private async createLockTableIfNotExists(
+    connection: DataSource,
+  ): Promise<void> {
+    await connection.query(
+      'CREATE TABLE IF NOT EXISTS "_lock" ("id" SERIAL NOT NULL, "status" int NOT NULL, PRIMARY KEY ("id"))',
+    );
+  }
+
+  /**
+   * Inserts a new lock into the database.
+   *
+   * @param {DataSource} connection The database connection to insert the lock into.
+   *
+   * @returns {Promise<void>} A promise that resolves when the lock has been inserted.
+   */
+  private async insertLock(connection: DataSource): Promise<void> {
+    await connection.query('INSERT INTO "_lock" (status) VALUES ($1);', [
+      migrationStatus.RUNNING,
+    ]);
+  }
+
+  /**
+   * Truncates the locks table in the database.
+   *
+   * @param {DataSource} connection The database connection to truncate the locks table.
+   *
+   * @returns {Promise<void>} A promise that resolves when the locks table has been truncated.
+   */
+  private async truncateLocks(connection: DataSource): Promise<void> {
+    await connection.query('TRUNCATE TABLE "_lock";');
+  }
+
+  /**
+   * Selects and retrieves locks from the database.
+   *
+   * @param {DataSource} connection The database connection to select locks from.
+   *
+   * @returns {Promise<Array<LockSchema>>} A promise that resolves to an array of LockSchema objects.
+   */
+  private async selectLock(connection: DataSource): Promise<Array<LockSchema>> {
+    return await connection.query('SELECT "id", "status" FROM "_lock";');
+  }
```

Comment:
Is there a reason [we can't use `pg_advisory_lock`](https://github.com/safe-global/safe-client-gateway/blob/ormMigrationAndLogger/src/datasources/db/v1/postgres-database.migration.hook.ts#L61-L69) here? We could use a similar pattern if so.

Reply:
I also tried using pg_advisory_lock, but since these locks are session-based, I found them to be quite unreliable. If the session is terminated, the lock is released, which could potentially cause issues in the database.

Reply:
I see. Can we add a comment explaining that?

Reply:
What about adding a `README` file to the module to explain this information?

Reply:
I am impartial. We don't tend to have localised `README`s but I have no strong opinion.

Reply:
Done https://github.com/safe-global/safe-client-gateway/pull/1988/commits/3f9292a5aed4148193ff80241f8e0afba86f902c